### PR TITLE
Update Renesas RX network interface port

### PR DESCRIPTION
**Pull request by tony-josi-aws**
Description
-----------
This PR updates the network interface for the Renesas RX devices (CC RX compiler), with respect to the multi network endpoint , IPv6 related FreeRTOS+TCP stack changes.

Test Steps
-----------
Tested the following using E2 studio [FreeRTOS+TCP demo for RX72N Envision Kit](https://github.com/tony-josi-aws/Renesas_RX72N_Envision_Kit_CC_FreeRTOS_TCP) project:

- DHCP
- IPv4 pings
- RA/ND [IPv6]
- DNSv4
    - `dns query4: 'aws.amazon.com' = 99.86.12.77 rc = 0`
- DNSv6
    - `dns query6: 'google.nl' = 2404:6800:4007:81d::2003 rc = 0`
- IPv6 pings (ICMPv6)

[Sample test results.](https://github.com/tony-josi-aws/Renesas_RX72N_Envision_Kit_CC_FreeRTOS_TCP/blob/master/readme.txt)

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- ~[ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.~ NA

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/include/FreeRTOS_IP_Private.h [hunk 1]
```diff
@@ -410,7 +410,7 @@ extern struct xNetworkInterface * pxNetworkInterfaces;
 
 /** @brief Macro calculates the number of elements in an array as a size_t. */
 #ifndef ARRAY_SIZE_X
-    #ifndef _WINDOWS_
+    #if !defined( _WINDOWS_ ) && !defined( __CCRX__ )
```
Comment (ActoryOu) [inline source/include/FreeRTOS_IP_Private.h:413]: What will happen without this?

File: source/portable/NetworkInterface/RX/NetworkInterface.c [hunk 2]
```diff
@@ -427,26 +483,35 @@ static int InitializeNetwork( void )
...
+    eth_ret = R_ETHER_Open_ZC2( ETHER_CHANNEL_0, ( const uint8_t * ) myethaddr, false );
 
     if( ETHER_SUCCESS != eth_ret )
     {
         return pdFALSE;
     }
 
-    eth_ret = R_ETHER_Open_ZC2( ETHER_CHANNEL_0, myethaddr, ETHER_FLAG_OFF );
+    #if ( ETHER_CHANNEL_MAX >= 2 )
+        eth_ret = R_ETHER_Open_ZC2( ETHER_CHANNEL_1, ( const uint8_t * ) myethaddr, false );
```
[the remark this answers is not in the document]
Comment (ActoryOu) [reply source/portable/NetworkInterface/RX/NetworkInterface.c:508]: Does RX board support more end-points than 2?

[the remark this answers is not in the document]
Comment (tony-josi-aws) [reply source/portable/NetworkInterface/RX/NetworkInterface.c]: RX72N's Ethernet Controller (ETHERC) supports 2 HW channels, but for the Envision kit boards only channel zero is connected to the PHY via MII, hence only 1 channel can be used. The `#if ( ETHER_CHANNEL_MAX >= 2 )` check was used to debug the network interface while testing, I have removed the check and the code. Good catch.
Thanks Actory.

File: source/include/FreeRTOS_IP_Private.h (hunk 1 above)
Comment (tony-josi-aws) [reply source/include/FreeRTOS_IP_Private.h:413]: The `ARRAY_SIZE_X` macro is used in the condition check of for loops, it seems like CC RX doesn't like multiple statements in the condition check part of the for loops (compilation error) similar to windows.

**Review by ActoryOu — DISMISSED** [review]

**Review by shubnil — DISMISSED** [review]

**Review by ActoryOu — APPROVED** [review]

**Review by shubnil — APPROVED** [review]

**Pull request merged**